3_harmonize.R

```diff
@@ -76,6 +78,25 @@ p3_targets_list <- list(
       } else {.}
     },
     map(p3_wqp_data_aoi_clean_grp)
+  ),
+  
+  # Summarize the number of records associated with each parameter,
+  # characteristic name, and harmonized units. The harmonized dataset
+  # can be summarized using any combination of columns by passing a
+  # different vector of column names in `grouping_cols`.
+  tar_target(
+    p3_wqp_records_summary_csv,
+    summarize_wqp_records(p3_wqp_data_aoi_clean_param, 
+                          grouping_cols = c('parameter', 
+                                            'CharacteristicName',
+                                            'ResultMeasure.MeasureUnitCode'),
+                          "3_harmonize/out/wqp_records_summary.csv")
+  ),
+  
+  # Save an output file containing pipeline metadata, including build time.
+  tar_target(
+    p3_pipeline_metadata_csv,
```

Comment:
This one is interesting because using the default of `NULL` means that the target itself doesn't have any dependencies in the pipeline. It's all alone, disconnected. This means that 1) it rebuilds early in the pipeline which means that it could incorrectly summarize the build time / targets, and 2) that it won't rebuild even if there are changes downstream. I think we could add the data summary as a dependency to force this to build at the end and if that changes, or we eliminate this and assume the user can look at that info afterwards.

Adding the data summary as a dependency could look like ...
```r
tar_target(
    p3_pipeline_metadata_csv, {
      # Adding the data output as a depends so that this runs *after* the data download completes
      p3_wqp_records_summary_csv
      summarize_targets(fileout = "3_harmonize/out/pipeline_summary.csv")
    }, format = 'file'
  )
```

It's all alone 😢 
![image](https://user-images.githubusercontent.com/13220910/181302450-350f4510-bd5e-49b3-950f-949c5924603e.png)

Reply:
Ah, great point. I've actually seen this before and you're right, it will just build that target before anything else. I've added two "pseudo-dependencies" to this metadata target to account for that. 

![network_screenshot](https://user-images.githubusercontent.com/8785034/181350020-6066e5bc-6add-4719-8140-a9afe56063e1.png)